File: 1001/1001.lua
-- 卡拉兹(Callatz)猜想

function calcCallatz(num, i)
	if num == 1 then return i end
	
	if num % 2 ~= 0 then
		num = num * 3 + 1
	end 
	num = num / 2
	return calcCallatz(num, i + 1)
end

-- print("please enter a integer: ")
integer = io.read("*num")
print(calcCallatz(integer, 0))

File: 1001/1001.py
# coding:utf-8
# 卡拉兹(Callatz)猜想

def calc_callatz(num, i = 0):
	if num == 1: 
		return i

	if num % 2 != 0: 
		num = num * 3 + 1

	num = num / 2
	return calc_callatz(num, i + 1)

print "please enter a integer: "
integer = input()
print calc_callatz(integer)